### ISSUE TITLE
H800 GRPO qwen3-8B 12k seq-len 训练一直OOM。

### ISSUE THREAD

**Issue by qianxiao1111** [OpenRLHF/OpenRLHF#1066]
以下是我的训练参数：
python3 -m openrlhf.cli.train_ppo_ray \
   --ref_num_nodes 1 \
   --ref_num_gpus_per_node 1 \
   --actor_num_nodes 1 \
   --actor_num_gpus_per_node 4 \
   --vllm_num_engines 1 \
   --vllm_tensor_parallel_size 2 \
   --vllm_gpu_memory_utilization 0.6 \
   --init_kl_coef 1e-3 \
   --kl_target 0.01 \
   --entropy_loss_coef 1e-4 \
   --gamma 1.0 \
   --use_kl_loss \
   --kl_estimator k3 \
   --advantage_estimator group_norm \
   --pretrain /data/model/Qwen3-8B \
   --agent_func_path /home/zjuici/zly/agentic-o1/rl_train/agent_func.py \
   --save_path /data/sft_outputs/rl_checkpoint/agentic_test \
   --ckpt_path /data/sft_outputs/rl_checkpoint/agentic_test \
   --save_hf_ckpt \
   --micro_train_batch_size 1 \
   --train_batch_size 4 \
   --micro_rollout_batch_size 32 \
   --rollout_batch_size 128 \
   --n_samples_per_prompt 2 \
   --max_epochs 2 \
   --prompt_max_len 4096 \
   --max_samples 1000000 \
   --generate_max_len 8192 \
   --zero_stage 3 \
   --bf16 \
   --actor_learning_rate 1e-6 \
   --critic_learning_rate 9e-6 \
   --prompt_data /home/zjuici/zly/modified_data_with_messages.jsonl \
   --input_key instruction \
   --label_key label \
   --apply_chat_template \
   --normalize_reward \
   --gradient_checkpointing \
   --vllm_sync_backend nccl \
   --enforce_eager \
   --vllm_enable_sleep \
   --use_liger_kernel \
   --adam_offload \
   --deepspeed_enable_sleep \
   --flash_attn \
   --grad_accum_dtype bf16 \
   --use_wandb $wandb_token

Actor使用4张卡单独加载， micro_batch_size 已经降到1了， adam_offload 和 ds3用上， 相同的底模使用llamafatory 进行sft训练的时候，12K seq_len + 6卡加载可以把per_device_batch_size 开到4。 
为什么对显存的开销会差异这么大？

**Comment by hijkzzz** [comment]
micro_rollout_batch_size 太大了

**Comment by qianxiao1111** [comment]
> micro_rollout_batch_size 太大了

1. 这部分rollout-samples 的kvcache 会在forward之前就加载到actor对应显卡的显存中吗？
2. 实际上， 我使用了更小的micro-rollout-batchsize 和 rollout batch size， 也还是会OOM
